add color for top 5 slowest times

SteveL-MSFT opened the pull request:
Fix build.ps1 to work on macOS by using `new-item` instead of mkdir
Update `Measure-Script` to find top 5 slowest lines and mark them as `Top` using a new member
Update formatting where `Top` is `$true` and `$PSStyle` exists, then use a red background

This makes it easy to see which top 5 lines are slowest

SteveL-MSFT committed:
add color for top 5 slowest times

ajansveld reviewed (changes requested):
The format expressions are backwards:
`$_.ExecutionTime -f '{0:mm\:ss\.fffffff}'`
should be
`'{0:mm\:ss\.fffffff}' -f $_.ExecutionTime`
It just happens to have the same result in this case.

ajansveld commented on src/PSProfiler.format.ps1xml:
```diff
@@ -73,8 +73,14 @@ FormatData for MeasureScript
-                                <FormatString>{0:mm\:ss\.fffffff}</FormatString>
+                                <ScriptBlock>
+                                    if ($_.Top -and $PSStyle -ne $null) {
+                                        $PSStyle.Background.Red + ($_.ExecutionTime -f '{0:mm\:ss\.fffffff}') + $PSStyle.Reset
```
```suggestion
                                        $PSStyle.Background.Red + ('{0:mm\:ss\.fffffff}' -f $_.ExecutionTime) + $PSStyle.Reset
```

ajansveld commented on src/PSProfiler.format.ps1xml:
```diff
@@ -73,8 +73,14 @@ FormatData for MeasureScript
+                                        $PSStyle.Background.Red + ($_.ExecutionTime -f '{0:mm\:ss\.fffffff}') + $PSStyle.Reset
+                                    }
+                                    else {
+                                        $_.ExecutionTime -f '{0:mm\:ss\.fffffff}'
```
```suggestion
                                        '{0:mm\:ss\.fffffff}' -f $_.ExecutionTime
```

SteveL-MSFT committed:
Update src/PSProfiler.format.ps1xml

Co-authored-by: ajansveld <ajansveld@gmail.com>

SteveL-MSFT committed:
Update src/PSProfiler.format.ps1xml

Co-authored-by: ajansveld <ajansveld@gmail.com>

IISResetMe committed:
Parameterize "Top X"

Loving the idea of highlight a Top 5, but I think the user should be able to override this behavior. 

This commit adds a new `-Top` parameter for that

IISResetMe commented:
Excellent feature idea! 

I've parameterized the `$Top` count - 5 might be too noisy when profiling small snippets, but not perhaps not comprehensive enough if you're looking at 100s (or 1000s 🙈 ) of lines of code.

Thanks @SteveL-MSFT @ajansveld for your contribution!
